src/Features/LanguageServer/Protocol/Handler/Diagnostics/WorkspacePullDiagnosticHandler.cs

```diff
@@ -120,24 +122,20 @@ async Task AddDocumentsFromProjectAsync(Project? project, ImmutableArray<string>
                     return;
                 }
 
-                // if the project doesn't necessarily have an open file in it, then only include it if the user has full
-                // solution analysis on.
-                if (!isOpen)
+                var isFSAOn = globalOptions.GetBackgroundAnalysisScope(project.Language) == BackgroundAnalysisScope.FullSolution;
+                var documents = ImmutableArray<Document>.Empty;
+                // If FSA is on, then add all the documents in the project.  Other analysis scopes are handled by the document pull handler.
```

Comment:
is that ok though?  won't this mean workspace diags and document diags for the same file might start conflicting with each other?

Reply:
yep, LSP open documents / razor documents are filtered out later